packages/next/build/utils.ts

```diff
@@ -579,9 +593,10 @@ export async function isPageStatic(
 
     const config = mod.config || {}
     return {
-      static: !hasStaticProps && !hasGetInitialProps,
+      static: !hasStaticProps && !hasGetInitialProps && !hasServerProps,
```

Comment:
I know this var makes the branch order above moot, but I still think it'd be nice to have correct order above.

Reply:
Updated